src/libs/ck-libs/NDMeshStreamer/NDMeshStreamer.h

```diff
@@ -144,21 +223,26 @@ class MeshStreamer : public CBase_MeshStreamer<dtype, RouterType> {
   bool useCompletionDetection_;
   CompletionDetector *detectorLocalObj_;
   virtual int copyDataItemIntoMessage(
-              MeshStreamerMessage<dtype> *destinationBuffer,
-              const void *dataItemHandle, bool copyIndirectly = false);
-  void insertData(const void *dataItemHandle, int destinationPe);
-  void broadcast(const void *dataItemHandle, int dimension,
-                 bool copyIndirectly);
+              MeshStreamerMessageV *destinationBuffer,
+              const DataItemHandle<dtype> *dataItemHandle, bool copyIndirectly = false);
+  virtual int copyDataIntoMessage(
+              MeshStreamerMessageV *destinationBuffer,
+              char *dataHandle, size_t size, CkArrayIndex index);
+  void insertData(const DataItemHandle<dtype> *dataItemHandle, int destinationPe);
+  void storeMessageData(int destinationPe,
```

Comment:
If I remember correctly, you said `storeMessageData` was used only at intermediate TRAM nodes unlike `storeMessage`. If this is true, can you rename it so that it reflects this difference?

Reply:
Done.